### PR TITLE
TextBox -> Inputs

### LINKED ISSUE #103: CodeMirror box on TextBox page is kinda broken
The cursor doesn't match the location you type in and every line is wrapped into `<pre>`:

https://user-images.githubusercontent.com/13383509/184684692-76b910ce-4fa4-4551-a779-fb3adcf441aa.mp4

For comparison, the script console:
Screenshot 2022-08-15 at 19 25 29

### LINKED ISSUE #251: Bad render on TextBox page for dynamic combobox
If you are at the [TextBox](https://weekly.ci.jenkins.io/design-library/TextBox/) page, enter a value into the "Title" field and hit enter, the page reloads and displays the output of the Jenkins error page, in plain HTML _inside_ the demo boxes:

https://user-images.githubusercontent.com/13383509/232249504-21e9e25b-d4f0-444e-9463-a2097b6c8dd1.mp4

### LINKED ISSUE #262: Improve CodeMirror documentation
### Describe your use-case which is not covered by existing documentation.

Currently, the last paragraph on https://weekly.ci.jenkins.io/design-library/TextBox/ mentions CodeMirror, but doesn't provide examples how to apply a codemirror-config or use a specific highlighter.

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

_No response_

### PR DESCRIPTION
**Pull request by mawinter69**
Rename TextBox to Inputs and add documentation for numbers and passwords.
Add more codemirror examples
Fixes #251, 
Fixes #262, 
Fixes #103

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/design-library-plugin/pull/281"></a>

### CODE REVIEW

File: src/main/resources/io/jenkins/plugins/designlibrary/Inputs/index.jelly [hunk 1]
```diff
@@ -48,6 +50,7 @@
     <p class="jdl-paragraph">${%combobox.description}</p>
 
     <form>
+      <button type="submit" disabled="true" style="display: none" aria-hidden="true"/>
```
Comment (mawinter69) [inline src/main/resources/io/jenkins/plugins/designlibrary/Inputs/index.jelly:53]: This prevents that the form is submitted when hitting enter in the field (#251)

File: src/main/resources/scss/abstracts/_overrides.scss [hunk 2]
```diff
@@ -22,9 +22,8 @@
 
 // Remove when code blocks follows user's theme
 // https://github.com/jenkinsci/design-library-plugin/issues/68
-pre {
+pre.jdl-component-code__code {
```
Comment (mawinter69) [inline src/main/resources/scss/abstracts/_overrides.scss:25]: This ensures that the important styles here are not applied to the codemirror styles (#103)

File: src/main/resources/io/jenkins/plugins/designlibrary/Inputs/index.jelly
Comment (mawinter69) [inline src/main/resources/io/jenkins/plugins/designlibrary/Inputs/index.jelly:1]: I tried to use `<l:tabPane>` but this breaks the rendering of the codemirror textareas which only start working properly after modifying the content.

**Review by timja — APPROVED** [review]
nice work

**Pull request merged**